source/MRMesh/MRMarchingCubes.cpp

```diff
@@ -349,38 +349,45 @@ Expected<TriMesh> volumeToMesh( const V& volume, const MarchingCubesParams& para
 
     VolumeIndexer indexer( volume.dims );
 
-    std::atomic<bool> keepGoing{ true };
-    auto mainThreadId = std::this_thread::get_id();
-    int lastSubMap = -1;
-
-    size_t threadCount = tbb::global_control::parameter( tbb::global_control::max_allowed_parallelism );
+    int threadCount = tbb::global_control::parameter( tbb::global_control::max_allowed_parallelism );
     if ( threadCount == 0 )
         threadCount = std::thread::hardware_concurrency();
     if ( threadCount == 0 )
         threadCount = 1;
 
-    const auto layerCount = (size_t)indexer.dims().z;
+    const int layerCount = indexer.dims().z;
     const auto layerSize = indexer.sizeXY();
     assert( indexer.size() == layerCount * layerSize );
 
     // more blocks than threads is recommended for better work distribution among threads since
     // every block demands unique amount of processing
-    const auto blockCount = std::min( layerCount, threadCount > 1 ? 4 * threadCount : 1 );
-    const auto layerPerBlockCount = (size_t)std::ceil( (float)layerCount / (float)blockCount );
+    const int blockCount = std::min( layerCount, threadCount > 1 ? 4 * threadCount : 1 );
+    const int layerPerBlockCount = (int)std::ceil( (float)layerCount / (float)blockCount );
     const auto blockSize = layerPerBlockCount * layerSize;
     assert( indexer.size() <= blockSize * blockCount );
 
     SeparationPointStorage sepStorage( blockCount, blockSize );
 
-    ParallelFor( size_t( 0 ), blockCount, [&] ( size_t blockIndex )
+    const auto callingThreadId = std::this_thread::get_id();
+    std::atomic<bool> keepGoing{ true };
+    
+    // avoid false sharing with other local variables
+    // by putting processedBits in its own cache line
+    constexpr int hardware_destructive_interference_size = 64;
+    struct alignas(hardware_destructive_interference_size) S
     {
-        auto & block = sepStorage.getBlock( blockIndex );
+        std::atomic<int> numProcessedLeyers{ 0 };
+    } s;
```

Comment:
Please provide better names for the class and the variable. You might also replace direct references, e.g.:
```c++
auto& numProcessedLayers = cacheLineStorage.numProcessedLayers;
```